### ISSUE TITLE
Query parameter rules

### ISSUE THREAD

**Issue by danherd** [cloudcreativity/laravel-json-api#128]
How can I only make the $queryRules in Validators.php apply to 'index' routes?

I'm trying to do a DELETE request and it's complaining about a query parameter that I want to be mandatory for an index request not being present.

**Comment by lindyhopchris** [comment]
Unfortunately this is a known bug that I'm sorting out in the next release.

As a tempory fix, in your validator provider overload the `queryRules` method and modify the query rules if it is a delete request. Something along these lines:

```php
protected function queryRules()
{
    if ("DELETE" === request()->getMethod()) {
        return [];
    }

    return parent::queryChecker();
}
```

**Comment by danherd** [comment]
Here's a slight improvement. I have this in a 'base' validators class:

```
protected function queryRules()
{
    if (explode('@', request()->route()->getActionName())[1] === 'index') {
        return parent::queryRules();
    }

    return [];
}
```

**Comment by lindyhopchris** [comment]
This is now fixed on the 1.0 branch.
